code/modules/vehicles/armored/__armored.dm

```diff
@@ -154,13 +156,50 @@
 	if(Adjacent(entering_thing, src))
 		return list(get_turf(entering_thing))
 
+/obj/effect/turret_toss_holder/
+	name = "turret explosion holder"
+	desc = "You shouldn't see this."
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/obj/effect/turret_toss_holder/proc/cleanup_turret_toss()
+	playsound(src, "sound/effects/metal_crash.ogg", 45)
+	
+	var/obj/effect/abstract/particle_holder/dirt_kickup_high = new(get_offset_target_turf(get_turf(src),7,4),/particles/dirt_kickup_large/deva)
+	var/obj/effect/abstract/particle_holder/dirt_kickup_aoe =  new(get_offset_target_turf(get_turf(src),7,4),/particles/dirt_kickup)
+	animate(src, alpha = 0, time = 6 SECONDS, easing = EASE_IN)
+	QDEL_IN(dirt_kickup_high, 10) //value taken from explosions.dm's duration var
+	QDEL_IN(dirt_kickup_aoe, 10)
+	QDEL_IN(src, 2 SECONDS)
+
 /obj/vehicle/sealed/armored/obj_destruction(damage_amount, damage_type, damage_flag)
 	playsound(get_turf(src), SFX_EXPLOSION_LARGE, 100, TRUE) //destroy sound is normally very quiet
 	new /obj/effect/temp_visual/explosion(get_turf(src), 7, LIGHT_COLOR_LAVA, FALSE, TRUE)
 	for(var/mob/living/nearby_mob AS in occupants + cheap_get_living_near(src, 7))
 		shake_camera(nearby_mob, 4, 2)
+
+	//play the turret toss
+	do_toss()
 	return ..()
 
+/obj/vehicle/sealed/armored/proc/do_toss()
```

Comment:
why not just inline this? theres no reason to have it be in a seperate proc since there only one case we want this to be happening in